Feature/os utils transfering

PeepoFrog opened the pull request:
New PR after fixes

PeepoFrog committed:
feat(osUtils): Transferred and refactored osUtils

Transferred and refactored osUtils from old repository

PeepoFrog committed:
feat(osUtils): Tests for osUtils

Added tests for osUtils

MrLutik commented on pkg/ryokaicommon/utils/os/os_test.go:
```diff
@@ -0,0 +1,216 @@
+	}
+}
+
+func TestCheckIfPortIsValid(t *testing.T) {
```
TestCheckIfPortIsValid ->  TestValidatePort

MrLutik commented on pkg/ryokaicommon/utils/os/os_test.go:
```diff
@@ -0,0 +1,216 @@
+		t.Error("Unable to delete directory in temporary folder")
+	}
+}
+
```
I would expect also TestValidateIP.

PeepoFrog committed:
fix(osUtils): Implementing Changes Requested in the Pull Request Review

TestCheckIfPortIsValid -> TestValidatePort, added test for ValidateIP, small spelling fix

MrLutik reviewed (approved):
Nicely done

andrsj commented on pkg/ryokaicommon/utils/os/os.go:
```diff
@@ -0,0 +1,187 @@
+// If os.Stat returns nil error, the file exists, and the function returns true.
+// If the error is of type os.IsNotExist, indicating the file does not exist, it returns false.
+// For any other type of error, it also returns false, treating inaccessible files as non-existent.
+func FileExist(path string) bool {
```
`PathExists` much better describes the function

andrsj commented on pkg/ryokaicommon/utils/os/os.go:
```diff
@@ -0,0 +1,187 @@
+func RunCommand(command string) ([]byte, error) {
+	args, err := shlex.Split(command)
+	if err != nil {
+		return []byte{}, fmt.Errorf("error when spiting cmd to array of args, err: %w", err)
```
`return nil, fmt.Errorf(...)` looks MUCH better

andrsj commented on pkg/ryokaicommon/utils/os/os.go:
```diff
@@ -0,0 +1,187 @@
+		return false
+	}
+
+	return !(port < 0 || port > 65535)
```
`return port >= 0 && port <= 65535` easier to read
